snapmaker/src/module/toolhead_laser.cpp

```diff
@@ -1025,16 +1025,34 @@ void ToolHeadLaser::InlineDisable() {
 
 void ToolHeadLaser::SetOutputInline(float power) {
   CheckFan(power);
-  planner.laser_inline.status.isEnabled = true;
 
   SetPower(power);
   planner.laser_inline.power = power_pwm_;
+  
+  if (power_pwm_ > 0)
+    planner.laser_inline.status.isEnabled = true;
+  else
+    planner.laser_inline.status.isEnabled = false;
```

Comment:
Do we want to set `isEnabled` false here? Won't that mess with the trapezoidal config? I'd think we'd wanna ramp down to 0 instead of immediately shutting off.

Reply:
> Won't that mess with the trapezoidal config?

No, because SetOutputInline is called at the time of planning when the gcode is parsed, and is queueing blocks for future execution. TurnOn_ISR is where the motion controller handles the trapezoids based on the commanded power level.

> Do we want to set isEnabled false here?

That's a fair question. To me there's a philosophical difference between enabled and 0 output vs disabled. But I suppose they manifest as the same behavior with 0 laser output.